src/lib/videotagging_extensions/detection.js

```diff
@@ -0,0 +1,203 @@
+                });
+                cb();
+            })  
+            }, (err) => {
+                console.info(`Error on ${self.detectionAlgorithmManager.getCurrentExportAlgorithm()} init:`, err);
+                cb(err)
+            });
+
+        function exportFrame(frameId, frameCanvas, canvasContext, frameExportCb) {
+            if (!self.visitedFrames.has(frameId)) {
+                return frameExportCb();
+            }
+            var frameTags = [];
+            //confirm that frame is tagged and that no tags are unlabeled 
+            var frameIsTagged = self.videotagging.frames.hasOwnProperty(frameId) && (self.videotagging.frames[frameId].length);
+            if (frameIsTagged && (self.videotagging.getUnlabeledRegionTags(frameId).length != self.videotagging.frames[frameId].length)) {
+                //genereate metadata from tags
+                self.videotagging.frames[frameId].map( (tag) => {
+                    if (!tag.tags[tag.tags.length-1]) {
+                        return console.log(`frame ${frameId} region ${tag.name} has no label`);
+                    }
+                    var stanW = self.videotagging.video.videoWidth/tag.width;
+                    var stanH = self.videotagging.video.videoHeight/tag.height;
+                    frameTags.push({
+                        class : tag.tags[tag.tags.length-1],
+                        x1 : parseInt(tag.x1 * stanW),
+                        y1 : parseInt(tag.y1 * stanH),
+                        x2 : parseInt(tag.x2 * stanW),
+                        y2 : parseInt(tag.y2 * stanH)
+                    });
+
+                });
+            }
+
+            //draw the frame to the canvas
+            var frameFileName = `${path.basename(self.videotagging.src, path.extname(self.videotagging.src))}_frame_${frameId}.jpg`;
+            var buf = self.canvasToJpgBuffer(frameCanvas, canvasContext);
+            self.detectionAlgorithmManager.exporter.exportFrame(frameFileName, buf, frameTags)
+                .then(()=>{
+                    frameExportCb();
+                }, (err) => {
+                    console.info('Error occured when trying to export frame', err);
+                    frameExportCb(err);
+                });
+        }
+    }
+
+    
+    //allows user to review model suggestions on a video
+    this.review = function(method, modelPath, reviewPath, cb) {
+        //if the export reviewPath directory does not exist create it and export all the frames then review
+        if (!fs.existsSync(reviewPath)) {
+            fs.mkdir(reviewPath, () =>{
+                this.mapVideo(saveFrame, "last").then( () => {
+                    reviewModel();
+                });
+            });
+        } else {
+            reviewModel();
+        }
+
+        function reviewModel() {
+            //run the model on the reviewPath directory
+            self.detectionAlgorithmManager.setReviewer(method, modelPath);
```

Comment:
I would connect the setReviewer + reviwer.reviewImagesFolder to the same function to make the abstraction more complete..